tests/puppeteer/pages/Install/install.js

```diff
@@ -0,0 +1,149 @@
+    await this.page.screenshot({path: 'test.png'});
+    const title = await this.getTextContent(selector);
+    await global.expect(title).to.equal(pageTitle);
+  }
+
+  /**
+   * Change install language in step 1
+   */
+  async setInstallLanguage() {
+    await this.page.select(this.languageSelect, global.INSTALL_LANGUAGE);
+  }
+
+  /**
+   * Go to next step
+   */
+  async nextStep() {
+    await this.page.waitForSelector(this.nextStepButton, {visible: true});
+    await this.page.click(this.nextStepButton, {waitUntil: 'domcontentloaded'});
+  }
+
+  /**
+   * Click on checkbox to agree on terms and conditions if its not checked already in step 2
+   */
+  async agreeToTermsAndConditions() {
+    const isChecked = await this.elementChecked(this.termsConditionsCheckbox);
+    if (!isChecked) {
+      await this.page.click(this.termsConditionsCheckbox);
+    }
+  }
+
+  /**
+   * Fill Information and Account Forms in step 4
+   */
+  async fillInformationForm() {
+    await this.page.type(this.shopNameInput, global.SHOPNAME);
+    await this.page.select(this.countrySelect, global.INSTALL_COUNTRY);
+    await this.page.type(this.firstNameInput, 'demo');
+    await this.page.type(this.lastNameInput, 'demo');
+    await this.page.type(this.emailInput, global.EMAIL);
+    await this.page.type(this.passwordInput, global.PASSWD);
+    await this.page.type(this.repeatPasswordInput, global.PASSWD);
+  }
+
+  /**
+   * Fill Database Form in step 5
+   */
+  async fillDatabaseForm() {
+    await this.page.click(this.dbLoginInput, {clickCount: 3});
+    await this.page.type(this.dbLoginInput, global.db_user);
+    await this.page.click(this.dbPasswordInput, {clickCount: 3});
+    await this.page.type(this.dbPasswordInput, global.db_passwd);
+    await this.page.click(this.tablePrefixInput, {clickCount: 3});
+  }
+
+  /**
+   * Check if database exist (if not, it will be created)
+   * and check if all set properly to submit form
+   */
+  async checkDatabaseConnected() {
+    await this.page.click(this.testDbConnectionButton);
+    // Create database 'prestashop' if not exist
+    if (await this.elementVisible(this.createDbButton)) {
+      await this.page.click(this.createDbButton);
+    }
+    await this.page.waitForSelector(this.dbResultCheckOk);
+  }
+
+  /**
+   * Check if prestashop is installed properly
+   */
+  async checkInstallationSuccessful() {
+    await this.page.waitForSelector(this.installationProgressBar, {visible: true});
+    await this.page.waitForSelector(this.finalStepPageTitle, {visible: true, timeout: 90000});
+    await this.checkStepTitle(this.finalStepPageTitle, this.finalStepEnTitle);
+  }
+
+  /**
+   * Go to FO after Installation and check that Prestashop logo exist
+   */
+  async goAndCheckFOAfterInstall() {
+    const FOPage = await this.openLinkWithTargetBlank(this.page, this.discoverFoButton);
+    await FOPage.bringToFront();
+    await FOPage.waitForSelector(this.FOLogo, {visible: true});
```

Comment:
the function `openLinkWithTargetBlank` check if a new target is opened and wait for 'body' selector.
if target is not opened => error 
if no selecor 'body' in new target => error
Adding response check on a new target will not be easy to implement (and I'm not sure we can).
Maybe checking another selector than logo after opening FO ?

Reply:
I think it's worth checking what happens to this test if the page is in error. I mean, even a 404 page has a `body` element...

Reply:
> I mean, even a 404 page has a body element...

yes, absolutely.

the problem is that we are forced to open link in new tab, and in that process, we loose the object page So : 
- we can't intercept request and response before clicking because we don't have the page.
- and after clicking, we must wait for new target to get the page, and after that it's too late to get response status.

The only solution I see is to reload the page, and I don't like it.

Reply:
I added some checks in FO : 
- User Info Icon in Header
- Cart Icon in Header